### ISSUE TITLE
On shutdown persistence

### ISSUE THREAD

**Issue by ConorOkus** [rust-bitcoin/rust-lightning#1191]
For network graph and scorer (neither of which have consistency requirements)

**Comment by jurvis** [comment]
hi @ConorOkus, I will like to take a look into working on this! 😄 

I just watched the Creating a Sample Lightning Node with LDK video you did with @valentinewallace and it looks like it is deliberate design for developers to [bring their own persistence solution](https://github.com/lightningdevkit/ldk-sample/blob/main/src/disk.rs#L76-L111) for `NetworkGraph` and `Scorer`. 

Is the thinking here to persist `NetworkGraph` and `Scorer` automatically like how we do for `ChannelManager` in `BackgroundProcessor`?

**Comment by TheBlueMatt** [comment]
> Is the thinking here to persist NetworkGraph and Scorer automatically like how we do for ChannelManager in BackgroundProcessor?

Yep! Note that we already have access to the network graph via `net_graph_msg_handler...network_graph()` which we use to call `remove_stale_channels` regularly, we just have to also persist it occasionally.

**Comment by jurvis** [comment]
@TheBlueMatt is there still a desire to work on persisting `Scorer`? I started playing around with seeing how we may implement it and it looks like we will need to add a parameter to `BackgroundProcessor::start` to have access to the scorer; unless there's something I'm missing?

**Comment by TheBlueMatt** [comment]
Yes, I think it makes sense to move that way. We currently do it by hand in the sample, but it'd be nice if the background processor had an `Option<>`al scorer and could persist that. We'll need to bound it by `Writeable` instead of `Score`, I believe, though we could do both but bindings will be sad about that (or we could do a supertrait that is `WriteableScore` that is auto-implemented for everything that implements `Score` and `Writeable`).

**Comment by jurvis** [comment]
@TheBlueMatt sounds good. I have a draft implementation that uses `LockableScore` but I'll work on creating the `WriteableScore` super-trait. Hope to have something by Sunday. Thanks for the context!

**Comment by tnull** [comment]
It seems this was fully addressed by #1376 and #1416. Feel free to reopen if I'm missing something.
